internals/models.py

```diff
@@ -319,20 +320,25 @@ def del_none(d):
       del_none(value)
   return d
 
-class DictModel(db.Model):
+class DictModel(ndb.Model):
   # def to_dict(self):
   #   return dict([(p, unicode(getattr(self, p))) for p in self.properties()])
 
+  def is_saved(self):
+    if self.key:
+      return True 
+    return False
+
   def format_for_template(self, add_id=True):
     d = self.to_dict()
     if add_id:
-      d['id'] = self.key().id()
+      d['id'] = self.key.integer_id()
     return d
 
   def to_dict(self):
     output = {}
 
-    for key, prop in self.properties().iteritems():
+    for key, prop in self._properties.iteritems():
```

Comment:
This includes obsolete fields from old entities in our staging datastore.  Trying to do getattr() with an obsolete field causes an exception.  As a fix, please add
```
    # Skip obsolete values that are still in our datastore
    if not hasattr(self, key):
      continue
```